### PR TITLE
lib/graphs: introduce PageRank computation on digraphs

### PR DESCRIPTION
**Pull request by Morriar**
Implement famous Google's `PageRank` algorithm for directed graphs.

Ping @ablondin

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by ablondin** [comment]
Nice! I'm happy to see an extension of the module `graph`.

Just a minor comment: the `==` method of `PRMap` is reflexive and symmetric, but it is not transitive, which is not recommended. It would be cleaner to have a `distance` method between two `PRMap`s and, if this distance is less than `0.001`, then the loop continues.

**Comment by privat** [comment]
A fast fix could be to rename `==` ad `is_aprox` or something (you could even add the 0.001 value as a parameter of the method)

File: lib/graphs/pagerank.nit [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		init
+		for k, v in other do self[k] = v
+	end
+
+	# Equality is based on `Float::is_approx` with precision `0.001`.
+	redef fun ==(o) do
+		if not o isa SELF then return false
+		for k1, v1 in self do
+			if not o.has_key(k1) then return false
+			if not 0.001.is_approx(v1, o[k1]) then return false
```
Comment (privat) [inline lib/graphs/pagerank.nit]: If I remember correctly, the correct way is `v1.is_approx(o[k1], 0.001)` the precision is the last parameter. http://nitlanguage.org/doc/stdlib/class_core__kernel__Float.html#core__kernel__Float__is_approx.definition

**Comment by Morriar** [comment]
Reroll with @ablondin and @privat remarks.

**Comment by privat** [comment]
+1

**Comment by ablondin** [comment]
+1

**Pull request merged**